datafusion/sql/src/expr/value.rs

```diff
@@ -357,3 +366,63 @@ fn has_units(val: &str) -> bool {
         || val.ends_with("nanosecond")
         || val.ends_with("nanoseconds")
 }
+
+/// Try to decode bytes from hex literal string.
+///
+/// None will be returned if the input literal is hex-invalid.
+fn try_decode_hex_literal(s: &str) -> Option<Vec<u8>> {
```

Comment:
```suggestion
fn try_decode_hex_literal(s: &str) -> Result<Vec<u8>> {
```

Reply:
I guess the error detail of the decode failure is not necessary because the original invalid hex string literal is enough for troubleshooting if decode fails.